cyral/resource_cyral_sidecar_listener.go

```diff
@@ -0,0 +1,422 @@
+		CreateContext: CreateResource(
+			ResourceOperationConfig{
+				Name:       "SidecarListenersResourceCreate",
+				HttpMethod: http.MethodPost,
+				CreateURL: func(d *schema.ResourceData, c *client.Client) string {
+					return fmt.Sprintf("https://%s/v1/sidecars/%s/listeners",
+						c.ControlPlane,
+						d.Get(SidecarIdKey).(string))
+
+				},
+				NewResourceData: func() ResourceData { return &SidecarListenerResource{} },
+				NewResponseData: func(_ *schema.ResourceData) ResponseData { return &CreateListenerAPIResponse{} },
+			}, ReadSidecarListenersConfig,
+		),
+		ReadContext: ReadResource(ReadSidecarListenersConfig),
+		UpdateContext: UpdateResource(
+			ResourceOperationConfig{
+				Name:       "SidecarListenersResourceUpdate",
+				HttpMethod: http.MethodPut,
+				CreateURL: func(d *schema.ResourceData, c *client.Client) string {
+					return fmt.Sprintf("https://%s/v1/sidecars/%s/listeners/%s",
+						c.ControlPlane,
+						d.Get(SidecarIdKey).(string),
+						d.Get(ListenerIdKey).(string))
+
+				},
+				NewResourceData: func() ResourceData { return &SidecarListenerResource{} },
+			}, ReadSidecarListenersConfig,
+		),
+		DeleteContext: DeleteResource(
+			ResourceOperationConfig{
+				Name:       "SidecarListenersResourceDelete",
+				HttpMethod: http.MethodDelete,
+				CreateURL: func(d *schema.ResourceData, c *client.Client) string {
+					return fmt.Sprintf("https://%s/v1/sidecars/%s/listeners/%s",
+						c.ControlPlane,
+						d.Get(SidecarIdKey).(string),
+						d.Get(ListenerIdKey).(string))
+				},
+			},
+		),
+		Schema: map[string]*schema.Schema{
+			ListenerIdKey: {
+				Description: "ID of the listener that will be bound to the sidecar.",
+				Type:        schema.TypeString,
+				ForceNew:    true,
+				Computed:    true,
+			},
+			SidecarIdKey: {
+				Description: "ID of the sidecar that the listener will be bound to.",
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+			},
+			RepoTypesKey: {
+				Description: "List of repository types that the listener supports. Currently limited to one repo type, eg [\"mysql\"].",
+				Type:        schema.TypeList,
+				Required:    true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			UnixListenerKey: {
+				Description: "Unix listener settings.",
+				Type:        schema.TypeSet,
+				Optional:    true,
+				// Notice the MaxItems: 1 here. This ensures that the user can only specify one this block.
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						FileKey: {
+							Description: "File in which the sidecar will listen for the given repository.",
+							Type:        schema.TypeString,
+							Required:    true,
+						},
+					},
+				},
+			},
+			MultiplexedKey: {
+				Description: "Multiplexed listener, defaults to not multiplexing (false). Not supported for all repository types.",
```

Comment:
Maybe a note such as "... for all repository types. See "repo_types" for details" would be good